### PR TITLE
Add dicom to sobp converter 

### LINKED ISSUE #384: add sobp.dat converters for Varian facilities
We already have pld2sobp.dat

For varian they may come as dicom or converted to .csv.

How to name these?

### PR DESCRIPTION
**Pull request by FardousR**
Closes #384

### CODE REVIEW

File: pymchelper/utils/plan.py [hunk 1]
```diff
@@ -0,0 +1,368 @@
...
+        self.f_divx = interp1d(energy, data[:, 6], kind=k)     # div x [rad]
+        self.f_divy = interp1d(energy, data[:, 7], kind=k)     # div y [rad]
+        self.f_covx = interp1d(energy, data[:, 8], kind=k)     # cov (x, x') [mm]
+        self.f_covy = interp1d(energy, data[:, 9], kind=k)     # cov (y, y') [mm]
+        self.data = data
+
+
+@dataclass
+class Spot:
+    x: float
```
Comment (grzanka) [inline pymchelper/utils/plan.py:99]: mention the units

File: pymchelper/utils/plan.py [hunk 2]
```diff
@@ -0,0 +1,368 @@
...
+    patient_name: str = ""  # Last name of patient
+    patient_initals: str = ""  # Initials of patient
+    patient_firstname: str = ""  # Last name of patient
+    plan_label: str = ""  #
+    plan_date: str = ""  #
+    nfields: int = 0
+    fields: list = []
+    bm: BeamModel = None  # optional beam model class
+
+    def load(self, file):
```
Comment (grzanka) [inline pymchelper/utils/plan.py:157]: I'd suggest to move parsing method outside the class
```
def load_plan(fname: str) -> Plan:
   p = Plan()
   (....)
   return p
```

**Review by grzanka — CHANGES_REQUESTED** [review]
first comments

File: pymchelper/utils/dicom2sobp.py [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+
+    psum = 0.0
+    for spot in plan.spots:
+        en = spot.energy  # nominal energy
+
+        pn = bm.f_ppmu(en) * spot.mu * 1e6  # number of primaries
+        psum += pn
+
+        s = ""
+        s += "{:.5f}".format(bm.f_e(en) / 1000.0)  # actual energy [GeV]
```
Comment (grzanka) [inline pymchelper/utils/dicom2sobp.py:176]: Consider using f-strings

```suggestion
        s += f"{bm.f_e(en) / 1000.0:.5f}"  # actual energy [GeV]
```

File: tests/test_plan.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+    exit_code = pymchelper.utils.radiotherapy.plan.main([str(input_file_path)])
+    assert exit_code == 0
+
+    # expected_output_file_path = Path(output_file)
+    # assert expected_output_file_path.exists()
+
+
+# @pytest.mark.parametrize("input_file_path", input_files)
+# def test_plan_bm(input_file_path: Path):
+#     """Test plan loading with beam model."""
```
Comment (nbassler) [inline tests/test_plan.py:57]: @grzanka need this :)

Comment (nbassler) [reply tests/test_plan.py:57]: `-b tests/res/pld/beam_model_generic.csv tests/res/dicom/RN.1.2.246.352.71.5.361940808526.37240.20200311150225.dcm`

**Review by grzanka — APPROVED** [review]

**Pull request merged**